test/e2e/autotls/auto_tls_test.go

```diff
@@ -63,65 +109,79 @@ func TestPerKsvcCertLocalCA(t *testing.T) {
 		t.Fatalf("Failed to create initial Service: %v: %v", names.Service, err)
 	}
 
-	cancel := turnOnAutoTLS(t, clients)
-	test.CleanupOnInterrupt(cancel)
-	defer cancel()
-
-	// Wait for certificate to be ready.
-	if err := waitForCertificateReady(t, clients, routenames.Certificate(objects.Route)); err != nil {
-		t.Fatalf("Error waiting for certificate to become ready: %v", err)
+	var certName string
+	if isNamespaceCertEnabled(t, clients) {
```

Comment:
In our goal state, the test should need no awareness of how auto-TLS happens, which is why I wanted stuff moved out of the test and into the test setup.  Can we move this too?